CMakeLists.txt

```diff
@@ -223,9 +242,12 @@ cmake_dependent_option(IREE_TARGET_BACKEND_LLVM_CPU_WASM "Enables WebAssembly in
 cmake_dependent_option(IREE_TARGET_BACKEND_METAL_SPIRV "Enables the 'metal-spirv' compiler target backend" ${IREE_TARGET_BACKEND_DEFAULTS} ${IREE_BUILD_COMPILER} OFF)
 cmake_dependent_option(IREE_TARGET_BACKEND_VULKAN_SPIRV "Enables the 'vulkan-spirv' compiler target backend" ${IREE_TARGET_BACKEND_DEFAULTS} ${IREE_BUILD_COMPILER} OFF)
 
+# Supported default target backends that are only available on certain
+# platforms.
+cmake_dependent_option(IREE_TARGET_BACKEND_CUDA "Enables the 'cuda' compiler target backend" ${IREE_CUDA_SDK_DOWNLOAD_AVAILABLE} ${IREE_BUILD_COMPILER} OFF)
```

Comment:
There's a comment above (that I can't comment on because GitHub is stupid)

```
#-------------------------------------------------------------------------------
# Compiler Target Options
# By default, all compiler targets supported by the current platform which do
# not require external deps are enabled by default. This can be changed with:
#   -DIREE_TARGET_BACKEND_DEFAULTS=OFF
#-------------------------------------------------------------------------------
```

I think we should probably update the "no external deps" wording. Additionally, I think we should still honor the `IREE_TARGET_BACKEND_DEFAULTS` (why is that plural?) for Cuda. So `CUDA_BACKEND_DEFAULT = IREE_TARGET_BACKEND_DEFAULTS && IREE_CUDA_SDK_DOWNLOAD_AVAILABLE`

Reply:
I reworked the options. I missed a few of those points during my own review earlier, so thanks for the nudge :D